README.md

```diff
@@ -192,6 +195,74 @@ This could be used like so: `crane provision service1`, `crane run -v databases`
 
 When using targets, it is also possible to cascade the commands to related containers. There are 2 different flags, `--cascade-affected` and `--cascade-dependencies`. In our example configuration above, when targeting the `mysql` container, the `apache` container would be considered to be "affected". When targeting the `apache` container, the `mysql` container would be considered as a "dependency". Both flags take a string argument, which specifies which type of cascading is desired, options are `volumesFrom`, `link`, `net` and `all`.
 
+## Templating
+You can also generate a textual output using  [Go templates](http://golang.org/pkg/text/template/). You need to specify your custom template file with `--template` (`-t`) flag. Depending on the optional `--output` (`-o`) flag's value the behaviour is different:
+
+* __Dump to `Stdout`:__ Leave it to dump the output to `os.Stdout`.
+```
+crane template -t path/to/myTemplate.tmpl
+```
+
+* __Dump to a file:__ Specify a file name to render the output in a file with that name.
+```
+crane template -t path/to/myTemplate.tmpl -o output/myOutput.txt
+```
+
+* __Generate a file per container:__ Specify a file format containing `%s` to use the template to generate a different output file for each container using the same template (substituting the `%s` by the corresponding container's name).
+```
+crane template -t path/to/myTemplate.tmpl -o output/%s.txt
+```
+
+The data object passed to the template is a `TemplateInfo` struct, or a `TemplateInfoForContainer` one if you want to generate a file per container. You can see here the structure of each struct or interface needed.
+
+```go
+type TemplateInfo struct {
```

Comment:
Yes, I agree. Particularly https://godoc.org/github.com/michaelsauter/crane/crane#TemplateInfo and https://godoc.org/github.com/michaelsauter/crane/crane#TemplateInfoForContainer